public/components/datasources/components/manage/accelerations/acceleration_details_flyout.tsx

```diff
@@ -24,20 +24,90 @@ import {
   onRefreshButtonClick,
   onDiscoverButtonClick,
   onDeleteButtonClick,
-} from '../accelerations/helpers/utils';
+} from './utils/acceleration_utils';
+import { coreRefs } from '../../../../../framework/core_refs';
+import { OpenSearchDashboardsResponse } from '../../../../../../../../src/core/server/http/router';
 
 export interface AccelerationDetailsFlyoutProps {
   acceleration: any;
 }
 
-export const AccelerationDetailsFlyout = (props: AccelerationDetailsFlyoutProps) => {
-  const { acceleration } = props;
+const getMappings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  console.log('getMappings index: ', index);
+  return coreRefs.dslService?.fetchFields(index);
+};
+
+const getSettings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  return coreRefs.dslService?.fetchSettings(index);
+};
+
+const getIndexInfo = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  return coreRefs.dslService?.fetchIndices(index);
+};
+
+const handlePromise = (
+  promise: Promise<OpenSearchDashboardsResponse> | undefined,
+  action: string
+) => {
+  return promise!
+    .then((data) => ({ status: 'fulfilled', action, data }))
+    .catch((error) => ({ status: 'rejected', action, error }));
+};
+
+export const AccelerationDetailsFlyout = ({
+  acceleration: selectedAcc,
+}: AccelerationDetailsFlyoutProps) => {
+  const { index, dataSourceName, acceleration } = selectedAcc;
+  const { flintIndexName } = acceleration;
   const [selectedTab, setSelectedTab] = useState('details');
   const tabsMap: { [key: string]: any } = {
     details: AccelerationDetailsTab,
     schema: AccelerationSchemaTab,
     sql_definition: AccelerationSqlTab,
   };
+  const [settings, setSettings] = useState<object>();
```

Comment:
Please define your settings, mappings and indexInfo types/interfaces instead of <object>

Reply:
I will open up a follow up to address all the types.

Reply:
follow up issue has been created: https://github.com/opensearch-project/dashboards-observability/issues/1534